phd-tests/tests/src/migrate.rs

```diff
@@ -396,3 +396,54 @@ async fn run_serial_history_test(
     )
     .await
 }
+
+#[phd_testcase]
+async fn migration_ensures_instance_metadata(ctx: &Framework) {
+    // Create a source instance, and fetch the instance metadata its metrics are
+    // generated with.
+    let mut source = ctx
+        .spawn_default_vm("migration_ensures_instance_metadata_source")
+        .await?;
+    let mut target = ctx
+        .spawn_successor_vm(
+            "migration_ensures_instance_metadata_target",
+            &source,
+            None,
+        )
+        .await?;
+    source.launch().await?;
+    source.wait_to_boot().await?;
+    let expected_metadata = source.vm_spec().metadata;
+    let source_metadata = source.get_spec().await?.properties.metadata;
+    assert_eq!(
+        expected_metadata, source_metadata,
+        "Source instance was not populated with the correct instance metadata"
+    );
+
+    // Migrate the instance to a new server, and refetch the metadata.
+    target
+        .migrate_from(&source, Uuid::new_v4(), MigrationTimeout::default())
+        .await?;
+    let expected_metadata = target.vm_spec().metadata;
+    let target_metadata = target.get_spec().await?.properties.metadata;
+    assert_eq!(
+        expected_metadata, target_metadata,
+        "Target instance was not populated with the correct instance metadata"
+    );
+
+    // Check that the source / target sled identifiers are different.
+    //
+    // In general, we expect that the sled serial and ID _should_ be different;
+    // the model and revision _may_ be different (if migrating to a different
+    // class of compute sled entirely); and that the project / silo IDs should
+    // be the same. That's the job of the control plane, though, not Propolis,
+    // so here we just check that the sled serial and UUIDs are different.
```

Comment:
I might reword this a bit: the sled ID and serial are expected to be different here because `spawn_successor_vm`'s behavior is to change them, and the target is expected to report its own identifiers and not the source's. (`spawn_successor_vm` behaves that way because that roughly emulates what the control plane would do, but it's technically fine for them to differ in this respect, so I might move the bulk of this comment to `refresh_sled_identifiers` and just describe things here in terms of `spawn_successor_vm`.)

Reply:
Done in 4e6ebde